### PR TITLE
Reduced Queries in Volunteer Profile by 25% 

### PR DESCRIPTION
**Pull request by Boorclark**
Issue #245

Reduced the number of queries in the Volunteer Profile page. Two queries in trainedParticipants() in participants.py were merged together as they both accessed the same table. Added a property to term objects called academicYearStartingTerm that returns the first term in the academic year they belong to and stores it in a cache. If the term calling the property is the starting term, it'll return and save itself to the cache. If there is something in the cache already, it will return the term object in the cache without accessing the database. Deleted the function that did the same thing as the property and modified the test suite accordinly to account for the change.

Files Changed:
- app/logic/participants.py
- app/logic/utils.py
- app/models/term.py
- tests/code/test_utils.py

### CODE REVIEW

File: app/models/term.py [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
     description = CharField()
     year = IntegerField()
     academicYear = CharField()
     isSummer = BooleanField(default=False)
     isCurrentTerm = BooleanField(default=False)
+
+    _cache = None
+
+    @property
+    def academicYearStartingTerm(self):
```
Comment (BrianRamsay) [inline app/models/term.py:12]: Give a brief description of what this function does. It's short, but it might not be clear at first glance

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
Looking pretty good. We'll have to wait until the time selector changes are in to see if that gets rid of those changes in here.

File: app/models/term.py (hunk 1 above)
Comment (Boorclark) [reply app/models/term.py:12]: The issue has been addressed. Resolving conversation.

**Comment by BrianRamsay** [comment]
A lot of comments to resolve. Let's take a look

**Comment by BrianRamsay** [comment]
This needs a merge from development and conflict resolution before review, @Boorclark , @Lyisus

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
This table was renamed in development

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**